addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/discovery/SamsungTvDiscoveryParticipant.java

```diff
@@ -52,14 +65,14 @@ public DiscoveryResult createResult(RemoteDevice device) {
             Map<String, Object> properties = new HashMap<>();
             properties.put(HOST_NAME, device.getIdentity().getDescriptorURL().getHost());
 
-            DiscoveryResult result = DiscoveryResultBuilder.create(uid)
-                    .withProperties(properties)
-                    .withLabel(getLabel(device))
-                    .build();
+            properties.putAll(RemoteControllerService.discover(device.getIdentity().getDescriptorURL().getHost()));
 
-            logger.debug("Created a DiscoveryResult for device '{}' with UDN '{}'",
+            DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                    .withLabel(getLabel(device)).build();
+
+            logger.debug("Created a DiscoveryResult for device '{}' with UDN '{}' and properties: {}",
                     device.getDetails().getModelDetails().getModelName(),
-                    device.getIdentity().getUdn().getIdentifierString());
+                    device.getIdentity().getUdn().getIdentifierString(), properties.toString());
```

Comment:
Don't call `.toString()` on parameter objects when logging. This will automatically be done by the logger (but only if the configured log level is high enough). This way the relatively expensive `toString` operations are only executed when strictly necessary.